File: 树/多重递归/检查平衡性.java
public class 检查平衡性 {
    class Solution {

        private int depth(TreeNode root){
            if(root == null) return 0;
            return Math.max(depth(root.left), depth(root.right))  + 1;
        }

        public boolean isBalanced(TreeNode root) {
            if(root == null) return true;
            if(Math.abs(depth(root.left) - depth(root.right)) > 1) return false;
            return isBalanced(root.left) && isBalanced(root.right);
        }
    }
}

File: 树/二叉查找树/检查平衡性.kt
import kotlin.math.abs
import kotlin.math.max


class Solution {
    private fun depth(root: TreeNode?):Int{
        if(root == null) return 0
        return max(depth(root.left) , depth(root.right) ) + 1
    }

    fun isBalanced(root: TreeNode?): Boolean {
        if(root == null) return true
        val left = depth(root.left)
        val right = depth(root.right)
        if(abs(left - right) > 1) return false
        else return isBalanced(root.left) && isBalanced(root.right)
    }
}